packages/story-editor/src/components/floatingMenu/elements/shared/focusTrapButton.js

```diff
@@ -0,0 +1,124 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+import { Button, BUTTON_TYPES } from '@googleforcreators/design-system';
+import { forwardRef } from '@googleforcreators/react';
+import { __, sprintf } from '@googleforcreators/i18n';
+import { v4 as uuidv4 } from 'uuid';
+
+const _FocusTrapButton = styled(Button).attrs({ type: BUTTON_TYPES.DEFAULT })`
+  background-color: transparent;
+  display: flex;
+  align-items: center;
+  justify-content: flex-start;
+  outline: none;
+  margin: 0;
+  padding: 0;
+  min-width: inherit;
+  ${({ styleOverride }) =>
+    css`
+      ${styleOverride}
+    `}
+`;
+_FocusTrapButton.propTypes = {
+  styleOverride: PropTypes.array,
+};
+
+export const handleReturnTrappedFocus = (e, buttonRef) => {
+  // only bubble up for moving focus
+  if (e.key === 'Tab') {
+    e.preventDefault();
+    buttonRef.current.focus();
+  } else {
+    e.stopPropagation();
+  }
+};
+
+/**
+ * Intended as a last resort effort to wrangle nested inputs within a <ContextMenu>. Freeform inputs should be avoided within menus if at all possible. Because this is a last resort, it is a shared element in the floating menu not the design-system.
+ * When it's necessary, wrap the freeform input in the <FocusTrapButton> and pass the associated handleReturnTrappedFocus as onKeyDown of the designated input.
+ * You will also need to give the input the imported CONTEXT_MENU_SKIP_ELEMENT class from design system. This tells the context menu to ignore the input in its focusableChildren so it relies on the FocusTrapButton as the focusable element for that input. This allows the freeform inputs to be typed into and arrowed around and then tab to get back into the menu.
+ *
+ * **Sample Usage**
+ * ```js
+ *
+ * function SomeComponent() {
+ * const buttonRef = useRef();
+ * const inputRef = useRef();
+ *
+ * <FocusTrapButton ref={buttonRef} inputRef={inputRef} inputLabel='Font size'>
+ * <input ref={inputRef} aria-label="Font size" onKeyDown={(e) => handleReturnTrappedFocus(e, buttonRef)} inputClassName={CONTEXT_MENU_SKIP_ELEMENT} />
+ * </FocusTrapButton>
+ * }
+ * ```
+ *
+ * @param {Object} props Props.
+ * @param {Object} props.inputRef ref associated with input element
+ * @param {string} props.inputLabel label of the associated input, will be used to create "Press Enter to edit [inputLabel]"
+ * @param {Array} props.styleOverride css to override button styles. Useful when an input has a specific style that adjusts the width, like color opacity.
+ * @param {Node} props.children children within wrapper
+ */
+
+const FocusTrapButton = forwardRef(function FocusTrapButton(
+  { inputRef, inputLabel, children, styleOverride },
+  ref
+) {
+  return (
+    <_FocusTrapButton
+      id={uuidv4()}
```

Comment:
I'm guessing this new id all the time fixes something since you're doing it in a couple of places. Just out of curiosity and forgetfulness, what's it solving?

Reply:
it is!  We need each focusable element to have an id otherwise the `menu` focus on arrow keydown gets janky and will send the focus back to the first element.  So by adding an id we can allow the various inputs and buttons and dropdowns in floating menus to take advantage of the context menu keyboard nav.

https://github.com/GoogleForCreators/web-stories-wp/pull/11273/files#diff-c62ae1c4616a0518137fda6727273417d4c0b4692e40e9cc7a7bdbd519713850R162-R176

Reply:
As far as memoizing it goes, it's just rendered the one time per use so it didn't seem worth the extra initialization.